src/validation/transaction.cairo

```diff
@@ -135,11 +133,193 @@ pub fn validate_absolute_locktime(
     check_threshold_result
 }
 
+/// Validate a transaction input. If relative locktime is enabled,
+/// ensure the input's locktime is respected.
+pub fn validate_relative_locktime(
+    tx: @Transaction, input_index: usize, block_height: u32, block_time: u32
```

Comment:
you can just have one argument `input: TxIn` that would also make your unit tests simpler

Reply:
so no need to collect  `tx: @Transaction,`, just the full input , `input: TxIn, block_height: u32, block_time: u32`

Reply:
you want something like this 

```
pub fn validate_relative_locktime(
    input: @TxIn, block_height: u32, block_height: u32, block_time: u32
) -> Result<(), ByteArray> 
```

then our test will pass the following input transaction 

```
 TxIn {
                    script: @from_hex(
                        "01091d8d76a82122082246acbb6cc51c839d9012ddaca46048de07ca8eec221518200241cdb85fab4815c6c624d6e932774f3fdf5fa2a1d3a1614951afb83269e1454e2002443047"
                    ),
                    sequence: 0xffffffff,
                    previous_output: OutPoint {
                        txid: hex_to_hash_rev(
                            "0437cd7f8525ceed2324359c2d0ba26006d92d856a9c20fa0241106ee5a597c9"
                        ),
                        vout: 0x00000000,
                        data: TxOut { value: 100, ..Default::default() },
                        block_height: Default::default(),
                        block_time: Default::default(),
                        is_coinbase: true,
                    },
                    witness: array![].span(),
                }

```
@m-kus

Reply:
Yes